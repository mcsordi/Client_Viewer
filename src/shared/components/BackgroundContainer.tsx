import React, { useContext } from 'react';
import { themeContext } from '../../contexts/ThemeContext/context.ts';

type IchildrenTheme = {
  children: React.ReactNode;
  theme?: boolean;
};

export const BackgroundContainer: React.FC<IchildrenTheme> = ({
  children,
  theme,
}) => {
  const { light } = useContext(themeContext);
  return (
    <section
      className={`w-full min-h-screen ${light || theme ? `bg-slate-200` : 'bg-zinc-900'}`}
    >
      {children}
    </section>
  );
};
