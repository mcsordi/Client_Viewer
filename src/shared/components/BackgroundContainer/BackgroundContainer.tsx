import React, { useContext } from 'react';
import { themeContext } from '../../../contexts/ThemeContext/context.ts';
import { toggle } from '../../../contexts/DrawerToggle/context.ts';

type IchildrenTheme = {
  children: React.ReactNode;
  theme?: boolean;
};

export const BackgroundContainer: React.FC<IchildrenTheme> = ({
  children,
  theme,
}) => {
  const { light } = useContext(themeContext);
  return (
    <section
      className={`flex w-full min-h-screen ${light || theme ? `bg-slate-200` : 'bg-zinc-900'}`}
    >
      {children}
    </section>
  );
};
