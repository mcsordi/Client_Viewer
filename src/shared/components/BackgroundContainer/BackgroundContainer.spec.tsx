import { render, screen } from '@testing-library/react';
import { BackgroundContainer } from './BackgroundContainer';
import React from 'react';

describe('<BackgroundContainer/>', () => {
  it('should render the background in the dark mode', () => {
    render(
      <BackgroundContainer theme={false}>
        <p>hola</p>
      </BackgroundContainer>,
    );
    const text = screen.getByText('hola');
    expect(text.parentElement).toHaveClass('bg-zinc-900');
  });
  it('should render the background in the light mode', () => {
    render(
      <BackgroundContainer theme={true}>
        <p>hola</p>
      </BackgroundContainer>,
    );
    const text = screen.getByText('hola');
    expect(text.parentElement).toHaveClass('bg-slate-200');
  });
  it('should match a snapshot', () => {
    render(
      <BackgroundContainer theme={true}>
        <p>hola</p>
      </BackgroundContainer>,
    );
    const text = screen.getByText('hola');
    expect(text.parentElement).toMatchSnapshot();
  });
});
