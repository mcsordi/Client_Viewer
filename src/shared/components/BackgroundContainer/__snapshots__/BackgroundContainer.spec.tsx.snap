// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundContainer/> should match a snapshot 1`] = `
<section
  class="flex w-full min-h-screen bg-slate-200"
>
  <p>
    hola
  </p>
</section>
`;
